ocular/src/utils.rs

```diff
@@ -0,0 +1,59 @@
+use http::Uri;
+
+pub(crate) fn parse_or_build_grpc_endpoint(input: &str) -> Result<String, http::Error> {
+    let mut uri = input.parse::<Uri>()?;
+
+    if uri.port().is_none() {
+        return Ok("".to_string());
+    }
+
+    if uri.scheme().is_none() {
+        let builder = Uri::builder();
+        uri = builder
+            .scheme("https")
+            // this is kind of weird
+            .authority(input)
+            .path_and_query("/")
+            .build()?;
+    }
+
+    Ok(uri.to_string())
+}
```

Comment:
I think this method is sketchy and would appreciate some feedback on it @EricBolten @tony-iqlusion